Adapt accountability results scopes for responsive view

Quentinchampenois opened the pull request:
#### :tophat: What? Why?

As a user when navigating on the accountability results on responsive display, scopes filter can expand on a large part of screen. 

This PR implements a scope selector for the responsive display and keeps the orignal scopes filter for the desktop display.

#### :pushpin: Related Issues
- Fixes https://meta.decidim.org/processes/roadmap/f/122/proposals/15782

#### Testing

* Seed application
* Visit : `http://localhost:3000/processes/[YOUR_PP]/f/15/results`
* Look at the scope filter on desktop display
* Then with inspector, toggle device toolbar (responsive mode on Chrome)
* See the scope selector 

### :camera: Screenshots
 
**Desktop view**

<img width="600" alt="Screenshot 2021-10-06 at 10 13 05" src="https://user-images.githubusercontent.com/26109239/136167986-19a392b6-22ac-4ecb-97fa-0426ce718b65.png">


**Responsive view**

<img width="550" alt="Screenshot 2021-11-24 at 18 05 32" src="https://user-images.githubusercontent.com/26109239/143303311-f3618f13-d3e7-4a0e-a119-5f13a5130e0d.png">

<img width="550" alt="Screenshot 2021-11-24 at 18 05 48" src="https://user-images.githubusercontent.com/26109239/143303367-c0175dd5-b08b-4069-b1a4-1725731ac610.png">


:hearts: Thank you!

andreslucena commented:
Hi @Quentinchampenois
Sorry for the delay on answering! 
Just a couple of details that we'd like to change:

1. Can you change the "Scopes" / "Filter by:" in the legend before clicking? I think that "Filter by:" / "Select a scope" should be a bit clearer. I mean here: 
![image](https://user-images.githubusercontent.com/717367/141436479-1cfc0c69-8883-434b-85b8-5d2a14facb5f.png)

2. On the modal itself, it seems like it's missing CSS. Could we reuse what we have in the processes list page? (/processes). 

![image](https://user-images.githubusercontent.com/717367/141439092-c8cc3a25-66cd-4a1d-bb69-242f96f2f806.png)

3. The title in the modal should be "Select a scope" (just like the processes modal) instead of "Filter by:" 

Thanks! Again, sorry for the delay, I thought that I've already reviewed this one 😅

Quentinchampenois commented:
Hi @andreslucena,

Thank you for your feedback, I will update it asap

Quentinchampenois commented:
Hi @andreslucena,

Sorry for the delay, I've implemented your suggestions in the last commit. As you can see in description, I've updated screenshots with the latest version. 
This PR is ready for review, thanks !

andreslucena commented:
> Sorry for the delay, I've implemented your suggestions in the last commit.

Great! Sorry for my side now for the delay 😄 

> As you can see in description, I've updated screenshots with the latest version.

I've removed the old ones, as they could cause confusion

I see just another small detail that I didn't catch before, sorry :S 

The problem that I see is that with these new changes we're not showing when there's a scope selected. For instance, if I select a scope called "California":

1. In desktop view is selected:

![image](https://user-images.githubusercontent.com/717367/145583205-ceca9819-a674-4edb-92d5-813465b9f551.png)

2. But in responsive view we've lost that:
![image](https://user-images.githubusercontent.com/717367/145583366-bca3dd7f-2ea8-434f-bde6-aae9ae853d81.png)

It should work just like in processes page, with a label that I can remove by clicking, and if I select multiple scopes it should filter by them (as far as I see this is already implemented in the backend accountability, at least is working if I change the URLs manually)

![image](https://user-images.githubusercontent.com/717367/145583675-378526c8-a709-41fc-b69f-d20c1c43c8db.png)


(Sorry again for the delay and for not catching this mini bug before)

Quentinchampenois commented:
Alright, in this case I think it is better to implement the `filter_form_for` used in processes page.

It means I should refactor a little this form just to be implementable in accountability  : 
* make the area selector optional for the accountability results and keep this one on processes page.
* add a way to update translations when calling the partial (by example) and keep the current ones as default.

What do you think @andreslucena ?

Quentinchampenois commented:
Maybe I didn't understand your request, attached is a screenshot of the scopes selector display in mobile format, is this the display we want for the mobile version? 

<img width="537" alt="Screenshot 2021-12-14 at 10 05 31" src="https://user-images.githubusercontent.com/26109239/145968141-baac6040-f41e-4478-bc49-d060d01f85e8.png">

Also, I think I found a bug on the scope selector. When you add scopes, they don't get deleted when you click on them. As you can see above, scopes labels don't have the same display if you refresh the page before adding a new one. Maybe I will open an issue for this behaviour (tested on `develop` branch and [Codegram's staging](https://staging.decidim.codegram.com/processes?filter%5Bscope_id%5D%5B%5D=7&filter%5Bscope_id%5D%5B%5D=3&filter%5Barea_id%5D=))

andreslucena commented:
> Alright, in this case I think it is better to implement the filter_form_for used in processes page.

That makes sense!

> According to your second screenshot, I understand that you want the web scope selector on mobile view (with the scope label below the selector).

I don't understand what's the web scope selector

> Maybe I didn't understand your request, attached is a screenshot of the scopes selector display in mobile format, is this the display we want for the mobile version?

Yes, that's what I had in mind, but giving it a second thought, with this change we're adding the multiple scope selection and to be consistent with the desktop view we should also change the picker there... As that's a bigger change of what we had initially in mind, I need to have the rest of @decidim/product team approval first 😄 

> Also, I think I found a bug on the scope selector. When you add scopes, they don't get deleted when you click on them. As you can see above, scopes labels don't have the same display if you refresh the page before adding a new one. Maybe I will open an issue for this behaviour (tested on develop branch and Codegram's staging)

Nice catch!! This selector is buggy for sure. I didn't notice before that one. By playing a bit, I found a couple of minor bugs, I've reported them at #8611

andreslucena commented:
> As that's a bigger change of what we had initially in mind, I need to have the rest of @decidim/product team approval first smile

I've talked with @carolromero and we agreed that the best course of action would be:

1. To implement the scope selector in desktop and mobile 
2. As the current selector design would be too weird to have in the accountability page (as this page has other kind of design) ideally it'd look with the styles that you've already implemented here:

<img width="550" alt="Screenshot 2021-11-24 at 18 05 32" src="https://user-images.githubusercontent.com/26109239/143303311-f3618f13-d3e7-4a0e-a119-5f13a5130e0d.png">

3. This would introduce the same bugs that we've detected (#8611) so if you can also fix them (in another PR) would be awesome!

--
Let me know if you can implement this @Quentinchampenois - this could be tricky (specially the 2 point regarding accountability particular design).

Quentinchampenois commented:
Hello @carolromero , @andreslucena

I agree with you, reuse the scope selector would be the best option for homogeneity and maintenance. I think it implies a light refactor of the filters view to be customizable (for example disable areas selector in the accountability case), what do you think ?  

I am sorry, for now it seems to be complicated for me to find the time to work more on this refactor at the moment. Maybe I can close this PR for now, and rework on the fix and propose a new PR for this enhancement when I can.

andreslucena commented:
> Maybe I can close this PR for now, and rework on the fix and propose a new PR for this enhancement when I can. 

Ok, I agree. I prefer to have only PRs that we're actively working on, so it's easier to keep focus.
